### PR TITLE
Fix `index.css` button-menu color overriding behavior

### LINKED ISSUE #427: [BUG] - Navbar theme text color not being respected
### Context

When opting for a darker color palette in the navbar, it's noticeable that the new button text color in `2024.7.1` does not respect the color option set in the `navbar_text_color` variable.

After a more in-depth validation, I was able to conclude that the central theme present in the `jhub_apps/static/css/index.css` conflicts in some instances (in this case in this button) with the settings present here:

https://github.com/nebari-dev/jhub-apps/blob/a55f8a6effdd3325c704f87263a20f2b091da152/jhub_apps/templates/style.css#L279-L292

# Error in the CSS 
Bellow an image where the variable color for the nav text is not being respected, resulting in this darker color:
Captura de Tela 2024-07-31 às 21 26 36

It might be abt troublesome to see, but the corresponding button where the username resides is being computed in the HTML page with the following `className`:
```css
MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary 
MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root 
MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium 
MuiButton-textSizeMedium button-menu css-lgmdx1
```

Note the presence of the `button-menu` class here. You can also see in the right-most corner that the `color: var(--navbar-text-color) !important;` is being neglected, and instead, the one above (seen below) is being used.
```css
#app-bar #toolbar .button-menu {
    color: #0f1015 !important;
    font-weight: 700;
}
````
This class, `button-menu,` actually comes from the CSS theme present in the index.css outlined above. It has an `!important` tag alongside the color. Because this is computed first, it automatically gains priority over all changes from the `MuiButton*` objects. 

Below is the same output (theme settings) with the fix in the index.css already in place:

# Fixed order
Captura de Tela 2024-07-31 às 21 27 17

You can reproduce the above behavior using the following theme template vars for example:
```python
jupyterhub_theme = {
    "hub_title": "Welcome to Nebari",
    "hub_subtitle": "your open source data science platform",
    "welcome": "Running in dev mode",
    "display_version": True,
    "primary_color": "#4f4173",
    "primary_color_dark": "#4f4173",
    "secondary_color": "#957da6",
    "secondary_color_dark": "#957da6",
    "accent_color": "#32C574",
    "accent_color_dark": "#32C574",
    "text_color": "#111111",
    "h1_color": "#652e8e",
    "h2_color": "#652e8e",
    "navbar_color": "#1c1d26",
    "navbar_text_color": "#f1f1f6",
    "navbar_hover_color": "#652e8e",
}
c.JupyterHub.template_vars = {
    **themes.DEFAULT_THEME,
    **jupyterhub_theme,
}
``` 

### Value and/or benefit

User settings being respected as expected

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by viniciusdc**
## Reference Issues or PRs

closes #427 

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: jhub_apps/static/css/index.css [hunk 1]
```diff
@@ -1 +1,843 @@
...
+    justify-content: flex-end
+}
+
+.button-group-item {
+    margin: .25rem;
+    list-style: none
+}
+
+#app-bar #toolbar .button-menu {
+    color: #0f1015;
```
Comment (viniciusdc) [inline jhub_apps/static/css/index.css:493]: This was the actual change, `!important` was removed

**Comment by viniciusdc** [comment]
due to my extensions in my editor, it also expanded the CSS config in a more readable way, let me know if that needs to be restore to original.

**Comment by krassowski** [comment]
I think this file is not meant to be edited by hand. While many projects use compilation at release time, this one appears to be committing compiled static assets to git repo. In that case the change likely needs to be done elsewhere and the static assets need to be recompiled. I guess this is why my PR https://github.com/nebari-dev/jhub-apps/pull/355 was not merged for over a month? I now force-pushed with static assets recompiled.

**Comment by jbouder** [comment]
I believe you're going to want to make your change here: https://github.com/nebari-dev/jhub-apps/blob/610e36c4151b466224642f5e380cc434bb30f514/ui/src/components/navigation/navigation.css#L2.

Then you'll need to do an `npm run build` from the ui directory and commit all changes

**Comment by viniciusdc** [comment]
> I believe you're going to want to make your change here:
> 
> https://github.com/nebari-dev/jhub-apps/blob/610e36c4151b466224642f5e380cc434bb30f514/ui/src/components/navigation/navigation.css#L2
> 
> .
> Then you'll need to do an `npm run build` from the ui directory and commit all changes

Nice, thanks @jbouder -- It was already late, so I didn't even noticed that file was an artifact of the build process XD, thanks for pointing out the right one

**Comment by viniciusdc** [comment]
@jbouder @aktech this is now correctly addressed thanks, ready for another look

**Review by jbouder — APPROVED** [review]
Makes sense, and works based on a quick test.

**Comment by viniciusdc** [comment]
@aktech may we get this merged and a new release cut? this is a blocker on Nebari 2024.7.1 using the latest jhub-apps

**Comment by jbouder** [comment]
> @aktech may we get this merged and a new release cut? this is a blocker on Nebari 2024.7.1 using the latest jhub-apps

@aktech , you’re up

**Comment by aktech** [comment]
I'll do another release, thanks for getting this in.

**Pull request merged**